tests/bulk/bulkwrite-update-008.phpt

```diff
@@ -0,0 +1,81 @@
+--TEST--
+MongoDB\Driver\BulkWrite::update() with sort option
+--SKIPIF--
+<?php require __DIR__ . "/../utils/basic-skipif.inc"; ?>
+<?php skip_if_not_live(); ?>
+<?php skip_if_server_version('<', '8.0'); ?>
+<?php skip_if_not_clean(); ?>
+--FILE--
+<?php
+require_once __DIR__ . "/../utils/basic.inc";
+
+class CommandLogger implements MongoDB\Driver\Monitoring\CommandSubscriber
+{
+    public function commandStarted(MongoDB\Driver\Monitoring\CommandStartedEvent $event): void
+    {
+        if ($event->getCommandName() !== 'update') {
+            return;
+        }
+
+        printf("update included sort: %s\n", json_encode($event->getCommand()->updates[0]->sort));
+    }
+
+    public function commandSucceeded(MongoDB\Driver\Monitoring\CommandSucceededEvent $event): void
+    {
+    }
+
+    public function commandFailed(MongoDB\Driver\Monitoring\CommandFailedEvent $event): void
+    {
+    }
+}
+
+$manager = create_test_manager();
+
+$bulk = new MongoDB\Driver\BulkWrite();
+$bulk->insert(['_id' => 1]);
+$bulk->insert(['_id' => 2]);
+$bulk->insert(['_id' => 3]);
+$manager->executeBulkWrite(NS, $bulk);
+
+MongoDB\Driver\Monitoring\addSubscriber(new CommandLogger);
+
+$bulk = new MongoDB\Driver\BulkWrite;
+$bulk->update(['_id' => ['$gt' => 1]], ['$set' => ['x' => 11]], ['sort' => ['_id' => 1]]);
+$manager->executeBulkWrite(NS, $bulk);
+
+$bulk = new MongoDB\Driver\BulkWrite;
+$bulk->update(['_id' => ['$gt' => 1]], ['x' => 22], ['sort' => ['_id' => -1]]);
```

Comment:
Did you want to test `replaceOne` here instead? This other method is in the [spec commit](https://github.com/mongodb/specifications/commit/da8e120514fd5e7a5934513f4fb7a63c7cadbb43). Otherwise I don't get the benefit of this 2nd update.

Reply:
This is actually testing the code path for `mongoc_bulk_operation_replace_one_with_opts`. PHPC's original bulk write API only provides a single `BulkWrite::update()` method, and we branch according to the structure of the second argument and `multi` option.

See: https://github.com/mongodb/mongo-php-driver/blob/1.20.1/src/MongoDB/BulkWrite.c#L452

This isn't relevant to the upcoming API for the bulkWrite _command_, which will have specific methods in PHPC.